diff --git a/aclocal.m4 b/aclocal.m4
index 98b9987..47b507a 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1,12 +1,3 @@
-dnl We require that everyone use exactly the same Autoconf version so that
-dnl the internal functions defined and used by the main configure script
-dnl match those expected by the fragments.
-m4_define([GLIBC_AUTOCONF_VERSION], [2.68])
-m4_if(m4_defn([AC_AUTOCONF_VERSION]), GLIBC_AUTOCONF_VERSION, [],
-      [m4_fatal(m4_flatten(
-Exactly version GLIBC_AUTOCONF_VERSION of Autoconf is required but you have
-m4_defn([AC_AUTOCONF_VERSION])
-), [63])])dnl
 dnl
 dnl We define the macro GLIBC_PROVIDES to do an AC_PROVIDE for each macro
 dnl which appears in configure.ac before the sysdep configure scripts are run.
