diff --git a/ChangeLog b/ChangeLog
index bb783ed..27c860f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -50,7 +50,7 @@
 	* elf/tst-relsort1mod2.c: New file.
 
 2014-01-04  Maxim Kuvyrkov  <maxim@kugelworks.com>
-	    Ondřej Bílka  <neleai@seznam.cz>
+	    Ondej Blka  <neleai@seznam.cz>
 
 	[BZ #15073]
 	* malloc/malloc.c (_int_free): Perform sanity check only if we
@@ -558,7 +558,7 @@
 	* stdlib/strtod_l.c (____STRTOF_INTERNAL): Clear lowest limb of
 	numerator after shifting it by one limb.
 
-2011-12-19  Rafael Ávila de Espíndola  <rafael.espindola@gmail.com>
+2011-12-19  Rafael vila de Espndola  <rafael.espindola@gmail.com>
 
 	* sysdeps/x86_64/fpu/bits/fenv.h (feraiseexcept): Define it only
 	under [__USE_EXTERN_INLINES].
diff --git a/aclocal.m4 b/aclocal.m4
index c2f6cea..4eedc7c 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -115,8 +115,16 @@ AC_CACHE_CHECK(whether $LD is GNU ld, libc_cv_prog_ld_gnu,
 gnu_ld=$libc_cv_prog_ld_gnu
 ])
 
-dnl Test a compiler option or options with an empty input file.
-dnl LIBC_TRY_CC_OPTION([options], [action-if-true], [action-if-false])
-AC_DEFUN([LIBC_TRY_CC_OPTION],
-[AS_IF([AC_TRY_COMMAND([${CC-cc} $1 -xc /dev/null -S -o /dev/null])],
-	[$2], [$3])])
+dnl Run a static link test with -nostdlib -nostartfiles.
+dnl LIBC_TRY_LINK_STATIC([code], [action-if-true], [action-if-false])
+AC_DEFUN([LIBC_TRY_LINK_STATIC],
+[cat > conftest.c <<EOF
+int _start (void) { return 0; }
+int __start (void) { return 0; }
+$1
+EOF
+AS_IF([AC_TRY_COMMAND([${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS -o conftest
+		       conftest.c -static -nostartfiles -nostdlib
+		       1>&AS_MESSAGE_LOG_FD])],
+      [$2], [$3])
+rm -f conftest*])
diff --git a/configure b/configure
index dc31bd7..d4e9ca4 100755
--- a/configure
+++ b/configure
@@ -6170,26 +6170,29 @@ else
     cat > conftest.c <<EOF
 int _start (void) { return 0; }
 int __start (void) { return 0; }
+
 int foo (void) { return 1; }
 int (*fp) (void) __attribute__ ((section (".init_array"))) = foo;
+
 EOF
-  if { ac_try='${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS -o conftest conftest.c
-		     -static -nostartfiles -nostdlib 1>&5'
+if { ac_try='${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS -o conftest
+		       conftest.c -static -nostartfiles -nostdlib
+		       1>&5'
   { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
   (eval $ac_try) 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; }
-  then
-    if $READELF -S conftest | fgrep INIT_ARRAY > /dev/null; then
-      libc_cv_initfini_array=yes
-    else
-      libc_cv_initfini_array=no
-    fi
+  test $ac_status = 0; }; }; then :
+  if $READELF -S conftest | fgrep INIT_ARRAY > /dev/null; then
+    libc_cv_initfini_array=yes
   else
     libc_cv_initfini_array=no
   fi
-  rm -f conftest*
+else
+  libc_cv_initfini_array=no
+fi
+rm -f conftest*
+
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $libc_cv_initfini_array" >&5
 $as_echo "$libc_cv_initfini_array" >&6; }
@@ -6202,22 +6205,23 @@ $as_echo_n "checking whether to use .ctors/.dtors header and trailer... " >&6; }
 if ${libc_cv_ctors_header+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-      libc_cv_ctors_header=yes
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-int
-main ()
-{
+    libc_cv_ctors_header=yes
+  cat > conftest.c <<EOF
+int _start (void) { return 0; }
+int __start (void) { return 0; }
 
-__attribute__ ((constructor)) void ctor (void) { puts("ctor"); }
-__attribute__ ((destructor))  void dtor (void) { puts("dtor"); }
+__attribute__ ((constructor)) void ctor (void) { asm (""); }
+__attribute__ ((destructor))  void dtor (void) { asm (""); }
 
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
+EOF
+if { ac_try='${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS -o conftest
+		       conftest.c -static -nostartfiles -nostdlib
+		       1>&5'
+  { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; }; then :
         if $READELF -WS conftest$ac_exeext | $AWK '
 	{ gsub(/\[ */, "[") }
 	$2 == ".ctors" || $2 == ".dtors" {
@@ -6239,8 +6243,7 @@ else
         as_fn_error $? "missing __attribute__ ((constructor)) support??" "$LINENO" 5
 
 fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
+rm -f conftest*
 
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $libc_cv_ctors_header" >&5
diff --git a/configure.in b/configure.in
index 29ce2da..6c45ffb 100644
--- a/configure.in
+++ b/configure.in
@@ -1448,36 +1448,29 @@ EOF
     fi
   fi
 
-  AC_CACHE_CHECK(for .preinit_array/.init_array/.fini_array support,
-		 libc_cv_initfini_array, [dnl
-  cat > conftest.c <<EOF
-int _start (void) { return 0; }
-int __start (void) { return 0; }
+AC_CACHE_CHECK(for .preinit_array/.init_array/.fini_array support,
+	       libc_cv_initfini_array, [dnl
+LIBC_TRY_LINK_STATIC([
 int foo (void) { return 1; }
 int (*fp) (void) __attribute__ ((section (".init_array"))) = foo;
-EOF
-  if AC_TRY_COMMAND([${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS -o conftest conftest.c
-		     -static -nostartfiles -nostdlib 1>&AS_MESSAGE_LOG_FD])
-  then
-    if $READELF -S conftest | fgrep INIT_ARRAY > /dev/null; then
-      libc_cv_initfini_array=yes
-    else
-      libc_cv_initfini_array=no
-    fi
+],
+  [if $READELF -S conftest | fgrep INIT_ARRAY > /dev/null; then
+    libc_cv_initfini_array=yes
   else
     libc_cv_initfini_array=no
-  fi
-  rm -f conftest*])
-  if test $libc_cv_initfini_array != yes; then
-    AC_MSG_ERROR([Need linker with .init_array/.fini_array support.])
-  fi
+  fi],
+  [libc_cv_initfini_array=no])
+])
+if test $libc_cv_initfini_array != yes; then
+  AC_MSG_ERROR([Need linker with .init_array/.fini_array support.])
+fi
 
-  AC_CACHE_CHECK(whether to use .ctors/.dtors header and trailer,
-		 libc_cv_ctors_header, [dnl
-    libc_cv_ctors_header=yes
-    AC_TRY_LINK([], [
-__attribute__ ((constructor)) void ctor (void) { puts("ctor"); }
-__attribute__ ((destructor))  void dtor (void) { puts("dtor"); }
+AC_CACHE_CHECK(whether to use .ctors/.dtors header and trailer,
+	       libc_cv_ctors_header, [dnl
+  libc_cv_ctors_header=yes
+  LIBC_TRY_LINK_STATIC([
+__attribute__ ((constructor)) void ctor (void) { asm (""); }
+__attribute__ ((destructor))  void dtor (void) { asm (""); }
 ],
 		[dnl
       AS_IF([$READELF -WS conftest$ac_exeext | $AWK '
