---
 malloc/obstack.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/malloc/obstack.c
+++ b/malloc/obstack.c
@@ -117,7 +117,7 @@
 /* A looong time ago (before 1994, anyway; we're not sure) this global variable
    was used by non-GNU-C macros to avoid multiple evaluation.  The GNU C
    library still exports it because somebody might use it.  */
-struct obstack *_obstack_compat;
+struct obstack *_obstack_compat = 0;
 compat_symbol (libc, _obstack_compat, _obstack, GLIBC_2_0);
 #  endif
 # endif
